R/hector.R

```diff
@@ -86,6 +91,11 @@ runscenario <- function(infile)
 newcore <- function(inifile, loglevel=0, suppresslogging=FALSE,
                     name="unnamed hector core")
 {
+    if (is.list(inifile)) {
+        tmp <- tempfile()
+        write_ini(inifile, tmp)
```

Comment:
We need to ensure that this temporary file is cleaned up once the core has been created (and that it is cleaned up if the function exits abnormally for some reason.)